index.js

```diff
@@ -108,10 +108,10 @@ Manager.prototype.reconnect = function(url){
   }
 
   this.socket = new Socket(url, opts);
-  this.socket.onopen = this.onOpen.bind(this);
-  this.socket.onclose = this.onClose.bind(this);
-  this.socket.onmessage = this.onMessage.bind(this);
-  this.socket.onerror = this.onError.bind(this);
+  this.socket.on('open', this.onOpen);
+  this.socket.on('close', this.onClose);
+  this.socket.on('message', this.onMessage);
```

Comment:
`onmessage`'s callback will assume an `Event` object, which won't be the case for this handler (this confusing impl detail is why I removed the on* properties :D)